src/controllers/Social.ts

```diff
@@ -94,6 +111,102 @@ export default class Social {
     });
   }
 
+  /**
+   * Grants permission for a set of keys and an account, specified by the `options.granteeAccountId`.
+   * The `options.signer` must be the owner of the set of keys.
+   * @param {IGrantWritePermissionOptions} options - the list of keys and the grantee account ID.
+   * @returns {Promise<transactions.Transaction>} a promise that resolves to a transaction that is ready to be signed
+   * and sent to the network.
+   * @throws {InvalidAccountIdError} if the grantee account ID or the account ID specified in the keys is invalid.
+   * @throws {KeyNotAllowedError} if account IDs specified in the keys does not match the signer (granter) account ID.
+   */
+  public async grantWritePermission({
+    blockHash,
+    granteeAccountId,
+    keys,
+    nonce,
+    publicKey,
+    signer,
+  }: IGrantWritePermissionOptions): Promise<transactions.Transaction> {
+    if (!validateAccountId(granteeAccountId)) {
+      throw new InvalidAccountIdError(
+        granteeAccountId,
+        `the grantee account id is not valid`
+      );
+    }
+
+    // validate the keys
+    keys.forEach((value) => {
+      const accountId = value.split('/')[0] || '';
+
+      if (!validateAccountId(accountId)) {
+        throw new InvalidAccountIdError(accountId);
+      }
+
+      // if the key does not belong to the signer (granter) it cannot give grant permission
+      if (accountId !== signer.accountId) {
+        throw new KeyNotAllowedError(
+          value,
+          `key "${value}" does not belong to granter "${signer.accountId}"`
+        );
+      }
+    });
+
+    return transactions.createTransaction(
+      signer.accountId,
+      utils.PublicKey.fromString(publicKey.toString()),
+      this.contractId,
+      nonce,
+      [
+        transactions.functionCall(
+          ChangeMethodEnum.GrantWritePermission,
```

Comment:
I saw this, but I think as the account ID and the public key is passed in the JavaScript function, I just used the account ID. Do you recommend I use the public key instead of the account ID?